metricbeat/module/kubernetes/util/metrics_cache.go

```diff
@@ -97,26 +106,27 @@ func (m *valueMap) GetWithDefault(name string, def float64) float64 {
 func (m *valueMap) Set(name string, val float64) {
 	m.Lock()
 	defer m.Unlock()
-	m.ensureCleanupWorker()
-	m.values[name] = value{val, now().Add(m.timeout).Unix()}
+	v := &value{value: val}
+	v.renew(m.timeout)
+	m.values[name] = v
 }
 
-func (m *valueMap) ensureCleanupWorker() {
-	if !m.running {
-		// Run worker to cleanup expired entries
-		m.running = true
-		go func() {
-			for {
-				sleep(m.timeout)
-				m.Lock()
-				now := now().Unix()
-				for name, val := range m.values {
-					if now > val.expires {
-						delete(m.values, name)
-					}
+func (m *valueMap) startWorkers(ctx context.Context) {
+	go func() {
+		for {
+			var now time.Time
```

Comment:
zero-value of `time.Time` is not _now_. I don't remember exactly but I think it was 0-0-1 or 1-1-1970. If you want current time you should use `now := time.Now()`

Reply:
You are right, but it was only the variable definition, the value was set later. I have changed this code in any case.